variables.tf

```diff
@@ -1,9 +1,444 @@
+  }
+}
+
+variable "sysdig_crn" {
+  type        = string
+  description = "Cloud Monitoring crn for COS bucket (Optional)"
+  default     = null
+}
+
+##############################################################################
+# COS bucket encryption variables
+##############################################################################
+
+variable "existing_kms_instance_guid" {
+  description = "The GUID of the Key Protect or Hyper Protect instance in which the key specified in var.kms_key_crn is coming from. Required if var.skip_iam_authorization_policy is false in order to create an IAM Access Policy to allow Key Protect or Hyper Protect to access the newly created COS instance."
+  type        = string
+  default     = null
+
+  validation {
+    condition     = var.existing_kms_instance_guid != null ? can(regex("^[0-9a-fA-F]{8}-([0-9a-fA-F]{4}-){3}[0-9a-fA-F]{12}$", var.existing_kms_instance_guid)) : true
+    error_message = "must be a valid cos instance id"
+  }
+}
+
+# variable "kms_encryption_enabled" {
+#   description = "Set as true to use KMS key encryption to encrypt data in COS bucket (only applicable when var.create_cos_bucket is true)."
+#   type        = bool
+#   default     = true
+# }
+
+# variable "kms_key_crn" {
+#   description = "CRN of the KMS key to use to encrypt the data in the COS bucket. Required if var.encryption_enabled and var.create_cos_bucket are true."
+#   type        = string
+#   default     = null
+# }
+
+##############################################################
+# Context-based restriction (CBR)
+##############################################################
+
+variable "bucket_cbr_rules" {
+  type = list(object({
+    description = string
+    account_id  = string
+    rule_contexts = list(object({
+      attributes = optional(list(object({
+        name  = string
+        value = string
+    }))) }))
+    enforcement_mode = string
+    tags = optional(list(object({
+      name  = string
+      value = string
+    })), [])
+    operations = optional(list(object({
+      api_types = list(object({
+        api_type_id = string
+      }))
+    })))
+  }))
+  description = "(Optional, list) List of CBR rules to create for the bucket"
+  default     = []
+  # Validation happens in the rule module
+}
+
+variable "instance_cbr_rules" {
+  type = list(object({
+    description = string
+    account_id  = string
+    rule_contexts = list(object({
+      attributes = optional(list(object({
+        name  = string
+        value = string
+    }))) }))
+    enforcement_mode = string
+    tags = optional(list(object({
+      name  = string
+      value = string
+    })), [])
+    operations = optional(list(object({
+      api_types = list(object({
+        api_type_id = string
+      }))
+    })))
+  }))
+  description = "(Optional, list) List of CBR rules to create for the instance"
+  default     = []
+  # Validation happens in the rule module
+}
+
+variable "skip_iam_authorization_policy" {
```

Comment:
do you want to maybe rename `skip_cos_kms_auth_policy` to ensure its clear which auth policy this relates to (perhaps there will be more auth polices needed in the future)